src/windows/PushwooshPluginProxy.js

```diff
@@ -7,18 +7,23 @@ var platform = require('cordova/platform');
 module.exports = {
 
     onDeviceReady: function (success, fail, config) {
-        this.service = new PushSDK.NotificationService.getCurrent(config[0].appid);
+        try {
+            this.service = new PushSDK.NotificationService.getCurrent(config[0].appid);
 
-        var startPushData = null;
+            var startPushData = null;
 
-        if (platform.activationContext && platform.activationContext.args) {
-            startPushData = platform.activationContext.args;
-        }
+            if (platform.activationContext && platform.activationContext.args) {
+                startPushData = platform.activationContext.args;
+            }
 
-        if (startPushData !== null)
-            PushSDK.NotificationService.handleStartPush(startPushData);
+            if (startPushData !== null)
+                PushSDK.NotificationService.handleStartPush(startPushData);
 
-        success();
+            success();

```

Comment:
Typically, it's dangerous to use large try/catches in JS, because any unhandled errors in any code called through this path will trigger the catch. For example, if there is any kind of error in the `success()` function here (i.e., the code that we've sent from our app), it will trigger the `fail(e)` callback in the catch here—even though it's not actually a failure for `onDeviceReady()`.

Can you get around this by using defensive null checks, etc. instead?

Reply:
I can leave it as it was, but then the fail parameter is unused.

Reply:
Limited it to just the one line that was throwing the `Class not registered` error.